Integration Test Flake

<!-- If you need to report a security issue with Knative, send an email to knative-security@googlegroups.com. -->
## In what area(s)?
E2E Testing

<!-- Remove the '> ' to select -->

<!--
Classifications:
> /kind good-first-issue
> /kind doc
> /kind cleanup
-->

In this [E2E Test run](https://prow.knative.dev/view/gcs/knative-prow/pr-logs/pull/knative_client/432/pull-knative-client-integration-tests/1181485419032743939) an error occurred which is unrelated to the change in the PR:

```
--- PASS: TestServiceOptions/create,_update_and_validate_service_with_annotations (12.40s)
--- FAIL: TestTrafficSplit (364.94s)
    --- PASS: TestTrafficSplit/tag_two_revisions_as_v1_and_v2_and_give_50-50%_share (39.85s)
    --- FAIL: TestTrafficSplit/ramp/up_down_a_revision_to_20%_adjusting_other_traffic_to_accommodate (18.39s)
        traffic_split_test.go:367: assertion failed: error is not nil: expecting to receive parts of length 2, got 1 string: ,echo1-qkwgb-3,100,true seprator: |
    --- PASS: TestTrafficSplit/tag_a_revision_as_candidate,_without_otherwise_changing_any_traffic_split (12.07s)
    --- PASS: TestTrafficSplit/tag_a_revision_as_candidate,_set_2%_traffic_adjusting_other_traffic_to_accommodate (13.04s) 
```

--- a/test/e2e/traffic_split_test.go
+++ b/test/e2e/traffic_split_test.go
@@ -17,7 +17,6 @@
 package e2e
 
 import (
-	"errors"
 	"fmt"
 	"strconv"
 	"strings"
@@ -54,8 +53,8 @@ func splitTargets(s, separator string, partsCount int) ([]string, error) {
 	s = strings.TrimSuffix(s, targetsSeparator)
 	parts := strings.Split(s, separator)
 	if len(parts) != partsCount {
-		return nil, errors.New(fmt.Sprintf("expecting %d targets, got %d targets "+
-			"targets: %s seprator: %s", partsCount, len(parts), s, separator))
+		return nil, fmt.Errorf("expecting %d targets, got %d targets "+
+			"targets: %s seprator: %s", partsCount, len(parts), s, separator)
 	}
 	return parts, nil
 }

--- a/test/e2e/traffic_split_test.go
+++ b/test/e2e/traffic_split_test.go
@@ -51,10 +51,11 @@ func newTargetFields(tag, revision string, percent int, latest bool) TargetField
 }
 
 func splitTargets(s, separator string, partsCount int) ([]string, error) {
-	parts := strings.SplitN(s, separator, partsCount)
+	s = strings.TrimSuffix(s, targetsSeparator)
+	parts := strings.Split(s, separator)
 	if len(parts) != partsCount {
-		return nil, errors.New(fmt.Sprintf("expecting to receive parts of length %d, got %d "+
-			"string: %s seprator: %s", partsCount, len(parts), s, separator))
+		return nil, errors.New(fmt.Sprintf("expecting %d targets, got %d targets "+
+			"targets: %s seprator: %s", partsCount, len(parts), s, separator))
 	}
 	return parts, nil
 }
@@ -362,7 +363,6 @@ func TestTrafficSplit(t *testing.T) {
 func (test *e2eTest) verifyTargets(t *testing.T, serviceName string, expectedTargets []TargetFields) {
 	out := test.serviceDescribeWithJsonPath(t, serviceName, targetsJsonPath)
 	assert.Check(t, out != "")
-	out = strings.TrimSuffix(out, targetsSeparator)
 	actualTargets, err := splitTargets(out, targetsSeparator, len(expectedTargets))
 	assert.NilError(t, err)
 	formattedActualTargets := formatActualTargets(t, actualTargets)

